Purchases/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -717,46 +740,6 @@ private extension PurchasesOrchestrator {
         deviceCache.setOfferingsCacheTimestampToNow()
     }
 
-    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
-    func purchase(
-        sk2Product: SK2Product,
-        discount: StoreProductDiscountType?
-    ) async -> Result<(CustomerInfo, Bool), Error> {
```

Comment:
This method was redundant, we didn't need it to return `Result`. Instead now there's only one `async throws`.